WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift

```diff
@@ -294,18 +298,28 @@ extension SiteDesignContentCollectionViewController: UITableViewDataSource {
         }
         cell.delegate = self
         cell.selectionStyle = UITableViewCell.SelectionStyle.none
-        cell.section = isLoading ? nil : sections[indexPath.row]
-        cell.isGhostCell = isLoading
+
         if isLoading {
+            cell.section = nil
+            cell.isGhostCell = true
             cell.ghostThumbnailSize = ghostThumbnailSize
+            cell.collectionView.allowsSelection = false
+        } else {
+            let section = sections[indexPath.row]
+            cell.section = section
+            cell.isGhostCell = false
+            cell.collectionView.allowsSelection = true
+            cell.horizontalScrollOffset = sectionHorizontalOffsets[section.categorySlug] ?? .zero
```

Comment:
I understand that this refactor aims at make things a bit clearer, but it also generates code duplication. How about we extract all of this in a private method that takes a boolean state (`isLoading`) as a parameter and configures the cell accordingly?

Reply:
> but it also generates code duplication

True, but that's not always a bad thing if it makes things clearer. 😄

> extract all of this in a private method that takes a boolean state (isLoading) as a parameter and configures the cell accordingly

Are you thinking of a private method in this class or the cell's class?

Thanks!

Reply:
Maybe the cell? In this case it might not be private if we need to call it from the view controllers, but would also address the other comment about duplication. What do you think? In any case, since this is a non-blocking discussion, I'm going to approve the PR.